### PR TITLE
Cyborg Rechargers material changes, part two

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
Continuation of #74770;
- Fixes some botched math due to the borg material storage datums using a proprietary value per sheet (500 vs 2000)
- Restocking is now enabled by default when a borg enters a charger. It can be disabled during the charge session, but will re-enable the next time the charger is used.
- Moves the Metal and Glass storage datums to a new shared subtype.
- Restocking speed scales with the manipulator tier of the recharger; By default, it is 1/8th the total metal/glass storage per cycle, and increases to 1/5th with a tier 4 manipulator.
## Why It's Good For The Game
These were changes I meant to make before the last PR was merged.
## Changelog
:cl:
fix: Fixed math on cyborg restocking
qol: borgs entering chargers now have the restocking feature enabled by default. It can be disabled with a right-click while charging if you don't want to eat the station's mats.
balance: Borg restocking speed now scales with the charger's manipulator tier.
/:cl:

### CODE REVIEW

File: code/game/machinery/rechargestation.dm [hunk 1]
```diff
@@ -111,20 +110,24 @@
 /obj/machinery/recharge_station/proc/toggle_restock(mob/user)
 	if(sendmats)
 		sendmats = FALSE
-		balloon_alert(user, "restocking from ore silo: disabled")
+		if(user)
+			balloon_alert(user, "restocking from ore silo: disabled")
```
Comment (JohnFulpWillard) [inline code/game/machinery/rechargestation.dm:114]: We talked about the balloon alerts a little but i think they outta be shortened or swapped for something like say(), we can't expect people to read the whole thing without spamming it

Comment (GuillaumePrata) [reply code/game/machinery/rechargestation.dm:114]: ``silo link: disabled/enabled`` looks short enough.

Comment (zxaber) [reply code/game/machinery/rechargestation.dm]: Replaced with say() for these four. The restocking messages of `+ 15000u iron` are still balloon alerts.

**Comment by ArcaneMusic** [comment]
So, overall I'm down with the changes both here and in part 1, but I want to use this PR to say the general platform for the material economy going forward, since that was the major point of contention in the previous part and I've been trapped under a car for the past 4 months so I haven't been able to clarify on many of these points.

*  **Sources of infinite resources like how this was before** (borgs creating metal and glass out of thin air), at a negligible to **near-nothing cost should not exist**. It limits not only the importance of what materials can be, as well as completely negates the importance of any material that currently exists within game.

 Eg: Back in the day at least, meat-material was applied to synthblood-steaks, which could be infinitely produced through chem factories, and as such could create an infinite quantity of meat-material, with an export value the same as iron. A consequence of this, is that it served as an infinite source of something that could be used to produce "datum-material shit" like tables, chairs, walls, floors, whatever. As a result, since we can produce an infinite amount of it, we can't make anything cool out of datum-material recipies that can be used for anything, since the items are literally free.

And then you throw shit like metal-gen into the mix and... ugh. This is why it took me a week to get to this.

Needless to say, this is a good step in the right direction. However, a genuine effort will need to be made by someone (likely me) to start to evaluate the mineral costs used on stated with regards to how much ore is mined by players on average within a 45 minute to 90 minute round (as this is the average round length that we have designed the previous economy around). This is why I hesitated upon seeing the first part of this PR. I believe to going forward we should attempt to modernize the costs of recipes ahead of removing more of the crazy random infinite sources first, and this way we'll ease some growing pains as we then mercilessly cut out the remaining low effort-infinite material resource-I am banging my head against a Sisyphean boulder style- design issues such as this.

Tl;dr: your fine, I'd like us to fix part of the cause going forward before jumping to part of the symptoms

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 2]
```diff
@@ -149,8 +149,8 @@ GLOBAL_LIST_INIT(metal_recipes, list ( \
 	tableVariant = /obj/structure/table
 	material_type = /datum/material/iron
 	matter_amount = 4
-	cost = 500
-	source = /datum/robot_energy_storage/iron
+	cost = MINERAL_MATERIAL_AMOUNT
```
Comment (ArcaneMusic) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:152]: I like this

**Review by ArcaneMusic — APPROVED** [review]

**Comment by JohnFulpWillard** [comment]
I'm so sorry for doing this to you but I just merged a rework to materials and it renamed the define you use, please pull from master

**Review by ArcaneMusic — APPROVED** [review]
